src/components/Form2DataSection.js

```diff
@@ -17,6 +17,14 @@ class Form2DataSection extends React.Component {
         this.toggleSendNotifications = this.toggleSendNotifications.bind(this);
         this.toggleUseServiceEmail = this.toggleUseServiceEmail.bind(this);
     }
+    componentWillReceiveProps(nextProps) {
+        if (nextProps == null || nextProps.data == null) { return; }        
+        this.setState({
+            useServiceEmail: nextProps.data.useServiceEmail,
+            sendNotifications: nextProps.data.sendNotifications,
+            showEmailAsMandatory: nextProps.data.useServiceEmail || nextProps.data.sendNotifications
+        });
```

Comment:
I don't understand why you need to set the state instead of just allowing the props to flow through